### ISSUE TITLE
cannot decode object of class (OSInAppMessageInternal) for key

### ISSUE THREAD

**Issue by spirit-jsb** [OneSignal/OneSignal-iOS-SDK#993]
SDK Version: 3.6.0

oh, when i run my project crash in there.

``` text
2021-09-09 09:30:56.071455+0800 Lit Live[42366:195012] *** Terminating app due to uncaught exception 'NSInvalidUnarchiveOperationException', reason: '*** -[NSKeyedUnarchiver decodeObjectForKey:]: cannot decode object of class (OSInAppMessageInternal) for key (NS.objects) because no class named "OSInAppMessageInternal" was found; the class needs to be defined in source code or linked in from a library (ensure the class is part of the correct target). If the class was renamed, use setClassName:forClass: to add a class translation mapping to NSKeyedUnarchiver'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff20422fba __exceptionPreprocess + 242
	1   libobjc.A.dylib                     0x00007fff20193ff5 objc_exception_throw + 48
	2   Foundation                          0x00007fff20745b31 -[NSCoder(Exceptions) __failWithExceptionName:errorCode:format:] + 0
	3   Foundation                          0x00007fff20745cab -[NSCoder(Exceptions) __failWithExceptionName:errorCode:format:] + 378
	4   Foundation                          0x00007fff207d2f09 _decodeObjectBinary + 644
	5   Foundation                          0x00007fff207d51e0 -[NSKeyedUnarchiver _decodeArrayOfObjectsForKey:] + 850
	6   Foundation                          0x00007fff2072a70c -[NSArray(NSArray) initWithCoder:] + 161
	7   Foundation                          0x00007fff207d376a _decodeObjectBinary + 2789
	8   Foundation                          0x00007fff207d262a _decodeObject + 152
	9   Foundation                          0x00007fff207d2515 -[NSKeyedUnarchiver decodeObjectForKey:] + 162
	10  Foundation                          0x00007fff207d1663 +[NSKeyedUnarchiver unarchiveObjectWithData:] + 79
	11  LitLiveUI                           0x000000010ca92df5 -[OneSignalUserDefaults getSavedCodeableDataForKey:defaultValue:] + 245
	12  LitLiveUI                           0x000000010ca23a54 -[OSMessagingController init] + 276
	13  LitLiveUI                           0x000000010ca23700 __39+[OSMessagingController sharedInstance]_block_invoke + 112
	14  libdispatch.dylib                   0x000000011220c74e _dispatch_client_callout + 8
	15  libdispatch.dylib                   0x000000011220dc9b _dispatch_once_callout + 66
	16  LitLiveUI                           0x000000010ca23665 +[OSMessagingController sharedInstance] + 197
	17  LitLiveUI                           0x000000010ca568c9 +[OneSignal setInAppMessageClickHandler:] + 121
	18  LitLiveUI                           0x000000010c44b356 $s9LitLiveUI15OneSignalServerC05setupdE013launchOptionsySDySo019UIApplicationLaunchI3KeyaypGSg_tF + 662
	19  Lit Live                            0x0000000106758f16 $s8Lit_Live17ExternalLibrariesC14setupOnesignal13launchOptionsySDySo019UIApplicationLaunchH3KeyaypGSg_tF + 70
	20  Lit Live                            0x0000000106758b4c $s8Lit_Live17ExternalLibrariesC8initiateyySo13UIApplicationC_SDySo0F16LaunchOptionsKeyaypGSgtF + 108
	21  Lit Live                            0x00000001065429c8 $s8Lit_Live11AppDelegateC11application_29didFinishLaunchingWithOptionsSbSo13UIApplicationC_SDySo0k6LaunchJ3KeyaypGSgtFTo + 168
	22  UIKitCore                           0x00007fff246bd6f1 -[UIApplication _handleDelegateCallbacksWithOptions:isSuspended:restoreState:] + 232
	23  UIKitCore                           0x00007fff246bf33b -[UIApplication _callInitializationDelegatesWithActions:forCanvas:payload:fromOriginatingProcess:] + 4046
	24  UIKitCore                           0x00007fff246c4d63 -[UIApplication _runWithMainScene:transitionContext:completion:] + 1237
	25  UIKitCore                           0x00007fff23ce62d4 -[_UISceneLifecycleMultiplexer completeApplicationLaunchWithFBSScene:transitionContext:] + 179
	26  UIKitCore                           0x00007fff246c1142 -[UIApplication _compellApplicationLaunchToCompleteUnconditionally] + 59
	27  UIKitCore                           0x00007fff246c14d1 -[UIApplication _run] + 898
	28  UIKitCore                           0x00007fff246c639c UIApplicationMain + 101
	29  Lit Live                            0x0000000106543938 main + 56
	30  libdyld.dylib                       0x00007fff2025abbd start + 1
)
libc++abi: terminating with uncaught exception of type NSException
*** Terminating app due to uncaught exception 'NSInvalidUnarchiveOperationException', reason: '*** -[NSKeyedUnarchiver decodeObjectForKey:]: cannot decode object of class (OSInAppMessageInternal) for key (NS.objects) because no class named "OSInAppMessageInternal" was found; the class needs to be defined in source code or linked in from a library (ensure the class is part of the correct target). If the class was renamed, use setClassName:forClass: to add a class translation mapping to NSKeyedUnarchiver'
terminating with uncaught exception of type NSException
CoreSimulator 776.1 - Device: iPhone 11 (342B097E-2098-4BC2-9A2C-419AFAF75FA0) - Runtime: iOS 14.5 (18E182) - DeviceType: iPhone 11
```

**Comment by emawby** [comment]
@spirit-jsb Interesting the class OSInAppMessageInternal is a renamed class that was released with version 3.7.0. This crash would occur if you had cached in app messages using OneSignal 3.7.0 then downgraded to a lower version such as 3.6.0. At any point did you install a version of your app with OneSignal 3.7.0?

**Comment by spirit-jsb** [comment]
oh, yes, in my dev code used OneSignal 3.7.0 and prod code used OneSignal 3.6.0. Sorrrry, pls close this issus. Thx so much.
